radix-engine/src/kernel/track.rs

```diff
@@ -587,47 +589,38 @@ impl<'s> FinalizingTrack<'s> {
         invoke_result: Result<Vec<InstructionOutput>, RuntimeError>,
         fee_summary: &mut FeeSummary,
         vault_ops: Vec<(TraceActor, VaultId, VaultOp)>,
+        application_events: Vec<(EventTypeIdentifier, Vec<u8>)>,
+        application_logs: Vec<(Level, String)>,
```

Comment:
Should we revert events in case of transaction failures?

Reply:
Hmm... since these events have already been costed and paid for, why should we revert them on commit failures?

Reply:
I was leaning more on them being discarded and ignored by the gateway for debugging reasons but I think that there might be better ways to solve the simulator experience.